src/libcrun/linux.c

```diff
@@ -1689,7 +1698,7 @@ libcrun_create_dev (libcrun_container_t *container, int devfd, int srcfd,
           cleanup_close int dirfd = -1;
           char *basename, *tmp;
 
-          buffer = xstrdup (device->path);
+          buffer = xstrdup (normalized_path);
           dirname = buffer;
 
           tmp = strrchr (buffer, '/');
```

Comment:
You can move this to the top where you are now almost open-coding `strdup()` + `strrchr()`.